src/Microsoft.DotNet.Build.Tasks/PreprocessFile.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class PreprocessFile : Task
+    {
+        [Required]
+        public string SourceFile { get; set; }
+
+        [Required]
+        public string OutputFile { get; set; }
+
+        public string[] Defines { get; set; }
+
+        public override bool Execute()
+        {
+            HashSet<string> defineSet = new HashSet<string>(Defines ?? new string[] { }, StringComparer.Ordinal);
```

Comment:
I don't think we want to enforce the policy that FEATURE_FOO == feature_foo here.  ResGen didn't either, FWIW.